spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpProjectAutoConfiguration.java

```diff
@@ -0,0 +1,50 @@
+package org.springframework.cloud.gcp.core.autoconfig;
+
+import java.util.logging.Logger;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.google.common.base.Strings;
+
+/**
+ * @author João André Martins
+ */
+@Configuration
+@EnableConfigurationProperties(GcpProject.class)
+public class GcpProjectAutoConfiguration {
+
+  private static final Logger LOGGER =
+      Logger.getLogger(GcpProjectAutoConfiguration.class.getName());
+
+  private static final String PROJECT_ID_ENV_VAR_NAME = "GCP_PROJECT_ID";
```

Comment:
Let's use `GOOGLE_CLOUD_PROJECT` as done by the [client libraries](https://github.com/GoogleCloudPlatform/google-cloud-java/#specifying-a-project-id).